eng/common/pipelines/templates/stages/archetype-typespec-emitter.yml

```diff
@@ -0,0 +1,452 @@
+          Copy-Item $(publishArtifactsPath)/emitter-package-lock.json $(Build.SourcesDirectory)/eng/ -Force
+        displayName: Copy emitter-package json files
+
+      - ${{ parameters.InitializationSteps }}
+
+      - template: /eng/common/pipelines/templates/steps/git-push-changes.yml
+        parameters:
+          BaseRepoOwner: azure-sdk
+          TargetRepoName: $(Build.Repository.Name)
+          BaseRepoBranch: $(branchName)
+          CommitMsg: Initialize repository for autorest build $(Build.BuildNumber)
+          WorkingDirectory: $(Build.SourcesDirectory)
+          ScriptDirectory: $(Build.SourcesDirectory)/eng/common/scripts
+          # To accomodate scheduled runs and retries, we want to overwrite any existing changes on the branch
+          PushArgs: --force
+
+      - task: PowerShell@2
+        displayName: Get generation job matrix
+        name: generate_matrix
+        inputs:
+          pwsh: true
+          filePath: ./eng/common/scripts/New-RegenerateMatrix.ps1
+          arguments: >
+            -OutputDirectory "$(Build.ArtifactStagingDirectory)"
+            -OutputVariableName matrix
+            -JobCount ${{ parameters.RegenerationJobCount }}
+            -MinimumPerJob ${{ parameters.MinimumPerJob }}
+            -OnlyTypespec ${{ parameters.OnlyGenerateTypespec }}
+          workingDirectory: $(Build.SourcesDirectory)
+
+      - template: /eng/common/pipelines/templates/steps/publish-1es-artifact.yml
+        parameters:
+          artifactName: matrix_artifacts
+          artifactPath: $(Build.ArtifactStagingDirectory)
+
+    - job: Generate
+      dependsOn: Initialize
+      strategy:
+        matrix: $[dependencies.Initialize.outputs['generate_matrix.matrix']]
+      variables:
+        matrixArtifactsPath: $(Pipeline.Workspace)/matrix_artifacts
+      steps:
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+        parameters:
+          Paths:
+          - "/*"
+          - "!SessionRecords"
+          
+      - download: current
+        displayName: Download pipeline artifacts
+
+      - ${{ parameters.InitializationSteps }}
+
+      - task: PowerShell@2
+        displayName: Call regeneration script
+        inputs:
+          pwsh: true
+          filePath: ./eng/common/scripts/Update-GeneratedSdks.ps1
+          arguments: >
+            -PackageDirectoriesFile "$(matrixArtifactsPath)/$(DirectoryList)"
+          workingDirectory: $(Build.SourcesDirectory)
+        continueOnError: true
+
+      - template: /eng/common/pipelines/templates/steps/git-push-changes.yml
+        parameters:
+          BaseRepoOwner: azure-sdk
+          TargetRepoName: $(Build.Repository.Name)
+          BaseRepoBranch: $(branchName)
+          CommitMsg: Update SDK code $(JobKey)
+          WorkingDirectory: $(Build.SourcesDirectory)
+          ScriptDirectory: $(Build.SourcesDirectory)/eng/common/scripts
+
+    - job: Create_PR
+      displayName: Create PR
```

Comment:
I didn't think we needed to create a PR any longer given everything is in the one repo. Am I missing a scenario where we would do this?

Reply:
CI checks post regeneration as well as regeneration for uptake as part of the release cycle on prod emitter builds

Reply:
In a prod release run, c# will want to build, test and publish the emitter to npmjs.org, then use the new emitter to regenerate all typespec clients in the repo and have a PR available that updates emitter-package.json and contains the client changes.

Reply:
Yes I can see that as part of the post publish job isn't this about validation in a PR context for a emitter update?

Reply:
This is a full lifecycle template used for CI, PR, release and uptake.